Auto update failing for ollama

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the [ollama](https://github.com/termux-user-repository/tur/blob/e59aefb67856807a52d6abe194fefa6576223df1/tur/ollama) package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="console">
INFO: Updating ollama [Current version: 0.3.14]
INFO: package being updated to 0.4.0.
INFO: Trying to build package.
INFO: Package ollama exists in tur repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building ollama for arch aarch64...
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13133  100 13133    0     0   297k      0 --:--:-- --:--:-- --:--:--  298k
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   2244      0 --:--:-- --:--:-- --:--:--  2245
gpg: Signature made Thu 07 Nov 2024 12:33:45 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://tur.kcubeterm.com/dists/tur-packages/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 16805  100 16805    0     0   164k      0 --:--:-- --:--:-- --:--:--  164k
Downloading https://tur.kcubeterm.com/dists/tur-packages/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100   659  100   659    0     0   5269      0 --:--:-- --:--:-- --:--:--  5269
gpg: Signature made Wed 06 Nov 2024 01:08:55 AM UTC
gpg:                using RSA key 43EEC3A2934343315717FF6F6A5C550C260667D1
gpg: Good signature from "Termux-user-repository (TUR AUTOBUILD key)" [ultimate]
Downloading https://tur.kcubeterm.com/dists/tur-packages/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 16805  100 16805    0     0   133k      0 --:--:-- --:--:-- --:--:--  133k
Downloading https://tur.kcubeterm.com/dists/tur-packages/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100   659  100   659    0     0   7581      0 --:--:-- --:--:-- --:--:--  7581
gpg: Signature made Wed 06 Nov 2024 01:08:55 AM UTC
gpg:                using RSA key 43EEC3A2934343315717FF6F6A5C550C260667D1
gpg: Good signature from "Termux-user-repository (TUR AUTOBUILD key)" [ultimate]
Downloading https://tur.kcubeterm.com/dists/tur-packages/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 16805  100 16805    0     0   221k      0 --:--:-- --:--:-- --:--:--  221k
Downloading https://tur.kcubeterm.com/dists/tur-packages/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100   659  100   659    0     0   8886      0 --:--:-- --:--:-- --:--:--  8886
gpg: Signature made Wed 06 Nov 2024 01:08:55 AM UTC
gpg:                using RSA key 43EEC3A2934343315717FF6F6A5C550C260667D1
gpg: Good signature from "Termux-user-repository (TUR AUTOBUILD key)" [ultimate]
Downloading https://tur.kcubeterm.com/dists/tur-packages/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 16805  100 16805    0     0   210k      0 --:--:-- --:--:-- --:--:--  210k
Downloading https://tur.kcubeterm.com/dists/tur-packages/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100   659  100   659    0     0   5550      0 --:--:-- --:--:-- --:--:--  5550
gpg: Signature made Wed 06 Nov 2024 01:08:55 AM UTC
gpg:                using RSA key 43EEC3A2934343315717FF6F6A5C550C260667D1
gpg: Good signature from "Termux-user-repository (TUR AUTOBUILD key)" [ultimate]
Downloading https://tur.kcubeterm.com/dists/tur-packages/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 16805  100 16805    0     0   174k      0 --:--:-- --:--:-- --:--:--  174k
Downloading https://tur.kcubeterm.com/dists/tur-packages/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100   659  100   659    0     0  10295      0 --:--:-- --:--:-- --:--:-- 10295
gpg: Signature made Wed 06 Nov 2024 01:08:55 AM UTC
gpg:                using RSA key 43EEC3A2934343315717FF6F6A5C550C260667D1
gpg: Good signature from "Termux-user-repository (TUR AUTOBUILD key)" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13319  100 13319    0     0  29989      0 --:--:-- --:--:-- --:--:-- 29930
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   2115      0 --:--:-- --:--:-- --:--:--  2119
gpg: Signature made Tue 05 Nov 2024 06:26:11 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13119  100 13119    0     0  34764      0 --:--:-- --:--:-- --:--:-- 34706
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   2065      0 --:--:-- --:--:-- --:--:--  2066
gpg: Signature made Wed 06 Nov 2024 12:41:06 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading dependency libopenblas@0.3.28 if necessary...
Found libopenblas in https://packages-cf.termux.dev/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.dev/apt/termux-main/pool/main/libo/libopenblas/libopenblas_0.3.28_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 2463k  100 2463k    0     0  6203k      0 --:--:-- --:--:-- --:--:-- 6205k
extracting libopenblas to /home/builder/.termux-build/_cache-aarch64...
Downloading dependency blas-openblas@0.3.28 if necessary...
Found blas-openblas in https://packages-cf.termux.dev/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.dev/apt/termux-main/pool/main/b/blas-openblas/blas-openblas_0.3.28_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   988  100   988    0     0   2683      0 --:--:-- --:--:-- --:--:--  2684
extracting blas-openblas to /home/builder/.termux-build/_cache-aarch64...
Cloning into '/home/builder/.termux-build/ollama/cache/tmp-checkout'...
Note: switching to '9d71bcc3e2a97c8e62d758450f43aa212346410e'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

~/.termux-build/ollama/cache/tmp-checkout ~/.termux-build/ollama/cache
~/.termux-build/ollama/cache
Downloading https://golang.org/dl/go1.23.2.linux-amd64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    80  100    80    0     0    883      0 --:--:-- --:--:-- --:--:--   888

100    75  100    75    0     0    346      0 --:--:-- --:--:-- --:--:--   346

100 70.2M  100 70.2M    0     0   128M      0 --:--:-- --:--:-- --:--:--  128M
go: downloading github.com/google/uuid v1.1.2
go: downloading golang.org/x/crypto v0.23.0
go: downloading google.golang.org/protobuf v1.34.1
go: downloading golang.org/x/sys v0.20.0
go: downloading github.com/containerd/console v1.0.3
go: downloading github.com/d4l3k/go-bfloat16 v0.0.0-20211005043715-690c3bdd05f1
go: downloading github.com/mattn/go-runewidth v0.0.14
go: downloading github.com/olekukonko/tablewriter v0.0.5
go: downloading github.com/spf13/cobra v1.7.0
go: downloading golang.org/x/exp v0.0.0-20231110203233-9a3e6036ecaa
go: downloading golang.org/x/term v0.20.0
go: downloading github.com/nlpodyssey/gopickle v0.3.0
go: downloading github.com/pdevine/tensor v0.0.0-20240510204454-f88f4562727c
go: downloading github.com/x448/float16 v0.8.4
go: downloading github.com/gin-gonic/gin v1.10.0
go: downloading golang.org/x/text v0.15.0
go: downloading golang.org/x/sync v0.3.0
go: downloading github.com/emirpasic/gods v1.18.1
go: downloading golang.org/x/image v0.14.0
go: downloading github.com/agnivade/levenshtein v1.1.1
go: downloading github.com/gin-contrib/cors v1.7.2
go: downloading github.com/rivo/uniseg v0.2.0
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/apache/arrow/go/arrow v0.0.0-20211112161151-bc219186db40
go: downloading github.com/chewxy/hm v1.0.0
go: downloading github.com/chewxy/math32 v1.10.1
go: downloading github.com/google/flatbuffers v24.3.25+incompatible
go: downloading go4.org/unsafe/assume-no-moving-gc v0.0.0-20231121144256-b99613f794b6
go: downloading gonum.org/v1/gonum v0.15.0
go: downloading gorgonia.org/vecf32 v0.9.0
go: downloading gorgonia.org/vecf64 v0.9.0
go: downloading github.com/gin-contrib/sse v0.1.0
go: downloading github.com/mattn/go-isatty v0.0.20
go: downloading golang.org/x/net v0.25.0
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/golang/protobuf v1.5.4
go: downloading github.com/xtgo/set v1.0.0
go: downloading github.com/go-playground/validator/v10 v10.20.0
go: downloading github.com/pelletier/go-toml/v2 v2.2.2
go: downloading github.com/ugorji/go/codec v1.2.12
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/gabriel-vasile/mimetype v1.4.3
go: downloading github.com/go-playground/universal-translator v0.18.1
go: downloading github.com/leodido/go-urn v1.4.0
go: downloading github.com/go-playground/locales v0.14.1
rm -rf /home/builder/.termux-build/ollama/src/build/linux/amd64/default /home/builder/.termux-build/ollama/src/build/linux/amd64/metal /home/builder/.termux-build/ollama/src/build/linux/amd64/cpu /home/builder/.termux-build/ollama/src/build/linux/amd64/cpu_avx /home/builder/.termux-build/ollama/src/build/linux/amd64/cpu_avx2
make -f make/Makefile.default
make[1]: Entering directory '/home/builder/.termux-build/ollama/src/llama'
GOARCH=amd64 go build -buildmode=pie "-ldflags=-w -s \"-X=github.com/ollama/ollama/version.Version=0.4.0-0-g9d71bcc\" \"-X=github.com/ollama/ollama/llama.CpuFeatures=\" " -trimpath   -o /home/builder/.termux-build/ollama/src/llama/build/linux-amd64/runners/cpu/ollama_llama_server ./runner
GOARCH=amd64 go build -buildmode=pie "-ldflags=-w -s \"-X=github.com/ollama/ollama/version.Version=0.4.0-0-g9d71bcc\" \"-X=github.com/ollama/ollama/llama.CpuFeatures="avx"\" " -trimpath -tags "avx" -o /home/builder/.termux-build/ollama/src/llama/build/linux-amd64/runners/cpu_avx/ollama_llama_server ./runner
GOARCH=amd64 go build -buildmode=pie "-ldflags=-w -s \"-X=github.com/ollama/ollama/version.Version=0.4.0-0-g9d71bcc\" \"-X=github.com/ollama/ollama/llama.CpuFeatures="avx,avx2"\" " -trimpath -tags "avx,avx2" -o /home/builder/.termux-build/ollama/src/llama/build/linux-amd64/runners/cpu_avx2/ollama_llama_server ./runner
# runtime/cgo
gcc_amd64.S:27:8: error: unknown token in expression
 pushq %rbx
       ^
gcc_amd64.S:27:8: error: invalid operand
 pushq %rbx
       ^
gcc_amd64.S:28:8: error: unknown token in expression
 pushq %rbp
       ^
gcc_amd64.S:28:8: error: invalid operand
 pushq %rbp
       ^
gcc_amd64.S:29:8: error: unknown token in expression
 pushq %r12
       ^
gcc_amd64.S:29:8: error: invalid operand
 pushq %r12
       ^
gcc_amd64.S:30:8: error: unknown token in expression
 pushq %r13
       ^
gcc_amd64.S:30:8: error: invalid operand
 pushq %r13
       ^
gcc_amd64.S:31:8: error: unknown token in expression
 pushq %r14
       ^
gcc_amd64.S:31:8: error: invalid operand
 pushq %r14
       ^
gcc_amd64.S:32:8: error: unknown token in expression
 pushq %r15
       ^
gcc_amd64.S:32:8: error: invalid operand
 pushq %r15
       ^
gcc_amd64.S:39:7: error: unknown token in expression
 movq %rdi, %rbx
      ^
gcc_amd64.S:39:7: error: invalid operand
 movq %rdi, %rbx
      ^
gcc_amd64.S:40:7: error: unknown token in expression
 movq %rdx, %rdi
      ^
gcc_amd64.S:40:7: error: invalid operand
 movq %rdx, %rdi
      ^
gcc_amd64.S:41:7: error: invalid token in expression
 call *%rsi
      ^
gcc_amd64.S:41:7: error: invalid operand
 call *%rsi
      ^
gcc_amd64.S:42:7: error: invalid token in expression
 call *%rbx
      ^
gcc_amd64.S:42:7: error: invalid operand
 call *%rbx
      ^
gcc_amd64.S:45:7: error: unknown token in expression
 popq %r15
      ^
gcc_amd64.S:45:7: error: invalid operand
 popq %r15
      ^
gcc_amd64.S:46:7: error: unknown token in expression
 popq %r14
      ^
gcc_amd64.S:46:7: error: invalid operand
 popq %r14
      ^
gcc_amd64.S:47:7: error: unknown token in expression
 popq %r13
      ^
gcc_amd64.S:47:7: error: invalid operand
 popq %r13
      ^
gcc_amd64.S:48:7: error: unknown token in expression
 popq %r12
      ^
gcc_amd64.S:48:7: error: invalid operand
 popq %r12
      ^
gcc_amd64.S:49:7: error: unknown token in expression
 popq %rbp
      ^
gcc_amd64.S:49:7: error: invalid operand
 popq %rbp
      ^
gcc_amd64.S:50:7: error: unknown token in expression
 popq %rbx
      ^
gcc_amd64.S:50:7: error: invalid operand
 popq %rbx
      ^
# runtime/cgo
gcc_amd64.S:27:8: error: unknown token in expression
 pushq %rbx
       ^
gcc_amd64.S:27:8: error: invalid operand
 pushq %rbx
       ^
gcc_amd64.S:28:8: error: unknown token in expression
 pushq %rbp
       ^
gcc_amd64.S:28:8: error: invalid operand
 pushq %rbp
       ^
gcc_amd64.S:29:8: error: unknown token in expression
 pushq %r12
       ^
gcc_amd64.S:29:8: error: invalid operand
 pushq %r12
       ^
gcc_amd64.S:30:8: error: unknown token in expression
 pushq %r13
       ^
gcc_amd64.S:30:8: error: invalid operand
 pushq %r13
       ^
gcc_amd64.S:31:8: error: unknown token in expression
 pushq %r14
       ^
gcc_amd64.S:31:8: error: invalid operand
 pushq %r14
       ^
gcc_amd64.S:32:8: error: unknown token in expression
 pushq %r15
       ^
gcc_amd64.S:32:8: error: invalid operand
 pushq %r15
       ^
gcc_amd64.S:39:7: error: unknown token in expression
 movq %rdi, %rbx
      ^
gcc_amd64.S:39:7: error: invalid operand
 movq %rdi, %rbx
      ^
gcc_amd64.S:40:7: error: unknown token in expression
 movq %rdx, %rdi
      ^
gcc_amd64.S:40:7: error: invalid operand
 movq %rdx, %rdi
      ^
gcc_amd64.S:41:7: error: invalid token in expression
 call *%rsi
      ^
gcc_amd64.S:41:7: error: invalid operand
 call *%rsi
      ^
gcc_amd64.S:42:7: error: invalid token in expression
 call *%rbx
      ^
gcc_amd64.S:42:7: error: invalid operand
 call *%rbx
      ^
gcc_amd64.S:45:7: error: unknown token in expression
 popq %r15
      ^
gcc_amd64.S:45:7: error: invalid operand
 popq %r15
      ^
gcc_amd64.S:46:7: error: unknown token in expression
 popq %r14
      ^
gcc_amd64.S:46:7: error: invalid operand
 popq %r14
      ^
gcc_amd64.S:47:7: error: unknown token in expression
 popq %r13
      ^
gcc_amd64.S:47:7: error: invalid operand
 popq %r13
      ^
gcc_amd64.S:48:7: error: unknown token in expression
 popq %r12
      ^
gcc_amd64.S:48:7: error: invalid operand
 popq %r12
      ^
gcc_amd64.S:49:7: error: unknown token in expression
 popq %rbp
      ^
gcc_amd64.S:49:7: error: invalid operand
 popq %rbp
      ^
gcc_amd64.S:50:7: error: unknown token in expression
 popq %rbx
      ^
gcc_amd64.S:50:7: error: invalid operand
 popq %rbx
      ^
# runtime/cgo
gcc_amd64.S:27:8: error: unknown token in expression
 pushq %rbx
       ^
gcc_amd64.S:27:8: error: invalid operand
 pushq %rbx
       ^
gcc_amd64.S:28:8: error: unknown token in expression
 pushq %rbp
       ^
gcc_amd64.S:28:8: error: invalid operand
 pushq %rbp
       ^
gcc_amd64.S:29:8: error: unknown token in expression
 pushq %r12
       ^
gcc_amd64.S:29:8: error: invalid operand
 pushq %r12
       ^
gcc_amd64.S:30:8: error: unknown token in expression
 pushq %r13
       ^
gcc_amd64.S:30:8: error: invalid operand
 pushq %r13
       ^
gcc_amd64.S:31:8: error: unknown token in expression
 pushq %r14
       ^
gcc_amd64.S:31:8: error: invalid operand
 pushq %r14
       ^
gcc_amd64.S:32:8: error: unknown token in expression
 pushq %r15
       ^
gcc_amd64.S:32:8: error: invalid operand
 pushq %r15
       ^
gcc_amd64.S:39:7: error: unknown token in expression
 movq %rdi, %rbx
      ^
gcc_amd64.S:39:7: error: invalid operand
 movq %rdi, %rbx
      ^
gcc_amd64.S:40:7: error: unknown token in expression
 movq %rdx, %rdi
      ^
gcc_amd64.S:40:7: error: invalid operand
 movq %rdx, %rdi
      ^
gcc_amd64.S:41:7: error: invalid token in expression
 call *%rsi
      ^
gcc_amd64.S:41:7: error: invalid operand
 call *%rsi
      ^
gcc_amd64.S:42:7: error: invalid token in expression
 call *%rbx
      ^
gcc_amd64.S:42:7: error: invalid operand
 call *%rbx
      ^
gcc_amd64.S:45:7: error: unknown token in expression
 popq %r15
      ^
gcc_amd64.S:45:7: error: invalid operand
 popq %r15
      ^
gcc_amd64.S:46:7: error: unknown token in expression
 popq %r14
      ^
gcc_amd64.S:46:7: error: invalid operand
 popq %r14
      ^
gcc_amd64.S:47:7: error: unknown token in expression
 popq %r13
      ^
gcc_amd64.S:47:7: error: invalid operand
 popq %r13
      ^
gcc_amd64.S:48:7: error: unknown token in expression
 popq %r12
      ^
gcc_amd64.S:48:7: error: invalid operand
 popq %r12
      ^
gcc_amd64.S:49:7: error: unknown token in expression
 popq %rbp
      ^
gcc_amd64.S:49:7: error: invalid operand
 popq %rbp
      ^
gcc_amd64.S:50:7: error: unknown token in expression
 popq %rbx
      ^
gcc_amd64.S:50:7: error: invalid operand
 popq %rbx
      ^
make[1]: *** [make/Makefile.default:27: /home/builder/.termux-build/ollama/src/llama/build/linux-amd64/runners/cpu/ollama_llama_server] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [make/Makefile.default:32: /home/builder/.termux-build/ollama/src/llama/build/linux-amd64/runners/cpu_avx/ollama_llama_server] Error 1
make[1]: *** [make/Makefile.default:37: /home/builder/.termux-build/ollama/src/llama/build/linux-amd64/runners/cpu_avx2/ollama_llama_server] Error 1
make[1]: Leaving directory '/home/builder/.termux-build/ollama/src/llama'
make: *** [Makefile:41: default] Error 2
llama/llama.go:3: running "make": exit status 2
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2024-11-07 01:05:56 UTC.<br>
Run ID: <a href="https://github.com/termux-user-repository/tur/actions/runs/11714545579">11714545579</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/tur/ollama/0001-cgo-flags.patch
+++ b/tur/ollama/0001-cgo-flags.patch
@@ -1,0 +1,13 @@
+--- a/llama/llama.go
++++ b/llama/llama.go
+@@ -36,8 +36,8 @@
+ #cgo linux CXXFLAGS: -D_GNU_SOURCE
+ #cgo linux,amd64 LDFLAGS: -L${SRCDIR}/build/Linux/amd64
+ #cgo linux,amd64 LDFLAGS: -L${SRCDIR}/build/Linux/amd64
+-#cgo linux,arm64 CFLAGS: -D__aarch64__ -D__ARM_NEON -D__ARM_FEATURE_FMA -D__ARM_FEATURE_MATMUL_INT8
+-#cgo linux,arm64 CXXFLAGS: -D__aarch64__ -D__ARM_NEON -D__ARM_FEATURE_FMA -D__ARM_FEATURE_MATMUL_INT8
++#cgo linux,arm64 CFLAGS: -D__aarch64__ -D__ARM_NEON
++#cgo linux,arm64 CXXFLAGS: -D__aarch64__ -D__ARM_NEON
+ #cgo linux,arm64 LDFLAGS: -L${SRCDIR}/build/Linux/arm64
+ #cgo linux,arm64,sve CFLAGS: -march=armv8.6-a+sve
+ #cgo linux,arm64,sve CXXFLAGS: -march=armv8.6-a+sve

--- a/tur/ollama/build.sh
+++ b/tur/ollama/build.sh
@@ -1,25 +1,21 @@
 TERMUX_PKG_HOMEPAGE=https://ollama.com/
-TERMUX_PKG_DESCRIPTION="Get up and running with large language models. "
+TERMUX_PKG_DESCRIPTION="Get up and running with large language models"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@termux-user-repository"
-TERMUX_PKG_VERSION="0.3.14"
+TERMUX_PKG_VERSION="0.4.0"
 TERMUX_PKG_SRCURL=git+https://github.com/ollama/ollama
-TERMUX_PKG_DEPENDS="libopenblas"
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_UPDATE_TAG_TYPE="latest-release-tag"
 TERMUX_PKG_BLACKLISTED_ARCHES="arm, i686"
 
-termux_step_pre_configure() {
-	termux_setup_golang
-	termux_setup_cmake
-	LDFLAGS+=" -fopenmp -static-openmp"
-}
-
 termux_step_make() {
-	export OLLAMA_CUSTOM_CPU_DEFS="-DGGML_BLAS=ON -DGGML_BLAS_VENDOR=OpenBLAS -DGGML_NATIVE=OFF -DGGML_OPENMP=ON -DGGML_AVX=OFF -DGGML_AVX2=OFF -DGGML_AVX512=OFF -DGGML_FMA=OFF -DGGML_F16C=OFF"
+	termux_setup_golang
+
+	export ARCH="$GOARCH"
+	export VERSION="$TERMUX_PKG_VERSION"
 	go generate './...'
-	go build
+	go build -trimpath -ldflags="-w -s -X=github.com/ollama/ollama/version.Version=$TERMUX_PKG_VERSION -X=github.com/ollama/ollama/server.mode=release"
 }
 
 termux_step_make_install() {

